service/counter/user.go

```diff
@@ -0,0 +1,224 @@
+	}
+	botStatus2Label := map[model.BotState]string{
+		model.BotInactive: "bot-inactive",
+		model.BotActive:   "bot-active",
+		model.BotPaused:   "bot-paused",
+	}
+	go func() {
+		for e := range hub.Subscribe(1, event.UserCreated, event.UserUpdated, event.BotCreated, event.BotStateChanged, event.BotDeleted).Receiver {
+			switch e.Topic() {
+			case event.UserCreated:
+				createdUser := e.Fields["user"].(*model.User)
+				if createdUser.Bot {
+					break
+				}
+				counter.userCounter[model.UserAccountStatusActive].inc()
+				counter.userTotalCounter.inc()
+				usersCounter.WithLabelValues("user-active").Inc()
+				usersCounter.WithLabelValues("user-total").Inc()
+				counter.userStatus[createdUser.ID] = model.UserAccountStatusActive
+
+			case event.UserUpdated:
+				userID := e.Fields["user_id"].(uuid.UUID)
+				var newStatusInInt int
+				db.Model(&model.User{}).Where("id = ?", userID).Select("status").First(&newStatusInInt)
+				newStatus := model.UserAccountStatus(newStatusInInt)
+				preStatus := counter.userStatus[userID]
+
+				counter.userCounter[preStatus].dec()
+				counter.userCounter[newStatus].inc()
+				usersCounter.WithLabelValues(userStatus2Label[preStatus]).Dec()
+				usersCounter.WithLabelValues(userStatus2Label[newStatus]).Inc()
+				counter.userStatus[userID] = newStatus
+
+			case event.BotCreated:
+				botState := e.Fields["bot"].(*model.Bot).State
+
+				counter.botCounter[botState].inc()
+				counter.botTotalCounter.inc()
+				usersCounter.WithLabelValues(botStatus2Label[botState]).Inc()
+				usersCounter.WithLabelValues("bot-total").Inc()
+				createdBotID := e.Fields["bot_id"].(uuid.UUID)
+				counter.botStatus[createdBotID] = botState
+
+			case event.BotStateChanged:
+				changedBotID := e.Fields["bot_id"].(uuid.UUID)
+				newStatus := e.Fields["state"].(model.BotState)
+				preStatus := counter.botStatus[changedBotID]
+
+				counter.botCounter[preStatus].dec()
+				counter.botCounter[newStatus].inc()
+				usersCounter.WithLabelValues(botStatus2Label[preStatus]).Dec()
+				usersCounter.WithLabelValues(botStatus2Label[newStatus]).Inc()
+				counter.botStatus[changedBotID] = newStatus
+
+			case event.BotDeleted:
+				deletedBotID := e.Fields["bot_id"].(uuid.UUID)
+				deletedBotState := counter.botStatus[deletedBotID]
+
+				counter.botCounter[deletedBotState].dec()
+				counter.botTotalCounter.dec()
+				usersCounter.WithLabelValues(botStatus2Label[deletedBotState]).Dec()
+				usersCounter.WithLabelValues("bot-total").Dec()
+				delete(counter.botStatus, deletedBotID)
+			}
+		}
+	}()
+	return counter, nil
+}
+
+// countUsers 指定したステータスのユーザーを取得し、c.userStatusに保存、c.userCounterにカウントします
+func (c *usersCounterImpl) countUsers(status model.UserAccountStatus, db *gorm.DB) error {
+	tmpUserCounter := userCounter{
+		count: 0,
+	}
+	var selectedStateUser []model.User
+	if err := db.
+		Model(&model.User{}).
+		Where("bot = ?", "0").
+		Where("status = ?", strconv.Itoa(status.Int())).
```

Comment:
`Where` に与える値は文字列以外にも対応しているはずなので, そのままで大丈夫だと思います.

それと, ひとつのwhere句にまとめてしまった方が良さそうに思えました.

```suggestion
		Where("bot = ? AND status = ?", false, status.Int()).
```